#! /bin/sh /usr/share/dpatch/dpatch-run
## Item1830-fix-validation-nonce.dpatch by  <sven@quad.home.org.au>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: When using embedded method, user is always redirected to the validation 
## DP: page, because the questionmark is never removed (removed by the JS in 
## DP: strikeone)

@DPATCH@

Index: foswiki/lib/Foswiki/Validation.pm
===================================================================
--- foswiki/lib/Foswiki/Validation.pm (revision 4494)
+++ foswiki/lib/Foswiki/Validation.pm (revision 4507)
@@ -166,4 +166,5 @@
     return 1 if ( $Foswiki::cfg{Validation}{Method} eq 'none' );
     return 0 unless defined $nonce;
+    $nonce =~ s/^\?// if ( $Foswiki::cfg{Validation}{Method} ne 'strikeone' );
     print STDERR "V: CHECK: $nonce\n" if TRACE;
     my $actions = $cgis->param('VALID_ACTIONS');

